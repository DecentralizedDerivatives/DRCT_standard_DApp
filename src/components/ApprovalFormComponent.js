import React from 'react';
import { Field } from 'redux-form';
import { InputGroup } from 'reactstrap';
import InputNumber from './InputNumber';

export const ApprovalFormComponent = ({
  handleSubmit,
  onSubmit,
  positionInfo
}) => {
  return (
    <div>
      <form onSubmit={handleSubmit(onSubmit)}>
      <p>{positionInfo.tokenType + ' ' + positionInfo.symbol + ': ' + positionInfo.balance + ' (' + positionInfo.date + ')'}</p>
        <br />
        <InputGroup>
          <Field
            name="tokenAmount"
            label="Token Amount"
            component={InputNumber}
            step={"0.00001"}
          />
        </InputGroup>
        <br />
        <div>
          <button
            type="submit"
            className="btn btn-primary"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default ApprovalFormComponent;
