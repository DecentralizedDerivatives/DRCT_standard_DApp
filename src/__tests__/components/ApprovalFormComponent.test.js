import ApprovalFormComponent from '../../components/ApprovalFormComponent';

describe('<ApprovalFormComponent />', () => {
  it('renders the component', () => {
    const handleSubmit = jest.fn();
    const store = initFixtureStore();
    const wrapper = shallow(
      <ApprovalFormComponent
        store={store}
        handleSubmit={handleSubmit}
        selectOptions={store.getState().user.userTokens}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
