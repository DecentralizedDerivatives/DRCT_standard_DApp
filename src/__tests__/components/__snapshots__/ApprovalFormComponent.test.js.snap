// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApprovalFormComponent /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApprovalFormComponent
    handleSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    selectOptions={Array []}
    store={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <InputGroup
          tag="div"
        >
          <Field
            component={[Function]}
            label="Select Token"
            name="token"
            options={Object {}}
          />
        </InputGroup>
        <br />
        <InputGroup
          tag="div"
        >
          <Field
            component={[Function]}
            label="Token Amount"
            name="tokenAmount"
            step="0.00001"
          />
        </InputGroup>
        <br />
        <div>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InputGroup
            tag="div"
          >
            <Field
              component={[Function]}
              label="Select Token"
              name="token"
              options={Object {}}
            />
          </InputGroup>,
          <br />,
          <InputGroup
            tag="div"
          >
            <Field
              component={[Function]}
              label="Token Amount"
              name="tokenAmount"
              step="0.00001"
            />
          </InputGroup>,
          <br />,
          <div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Select Token"
              name="token"
              options={Object {}}
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Select Token",
              "name": "token",
              "options": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Token Amount"
              name="tokenAmount"
              step="0.00001"
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Token Amount",
              "name": "tokenAmount",
              "step": "0.00001",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <InputGroup
            tag="div"
          >
            <Field
              component={[Function]}
              label="Select Token"
              name="token"
              options={Object {}}
            />
          </InputGroup>
          <br />
          <InputGroup
            tag="div"
          >
            <Field
              component={[Function]}
              label="Token Amount"
              name="tokenAmount"
              step="0.00001"
            />
          </InputGroup>
          <br />
          <div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InputGroup
              tag="div"
            >
              <Field
                component={[Function]}
                label="Select Token"
                name="token"
                options={Object {}}
              />
            </InputGroup>,
            <br />,
            <InputGroup
              tag="div"
            >
              <Field
                component={[Function]}
                label="Token Amount"
                name="tokenAmount"
                step="0.00001"
              />
            </InputGroup>,
            <br />,
            <div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </div>,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Field
                component={[Function]}
                label="Select Token"
                name="token"
                options={Object {}}
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "label": "Select Token",
                "name": "token",
                "options": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Field
                component={[Function]}
                label="Token Amount"
                name="tokenAmount"
                step="0.00001"
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "label": "Token Amount",
                "name": "tokenAmount",
                "step": "0.00001",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
